`-Wreturn-std-move` compiler warnings from `assignable.hpp`

See, e.g., [this log](https://github.com/scipy/scipy/actions/runs/5534254243/jobs/10098953978?pr=18631) from SciPy's macOS x86-64 wheel builds:
```
    [812/1598] Compiling C++ object scipy/stats/_stats_pythran.cpython-310-darwin.so.p/meson-generated_..__stats_pythran.cpp.o
    In file included from scipy/stats/_stats_pythran.cpp:1:
    In file included from ../../../../../../../private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-build-env-yj77gq5q/overlay/lib/python3.10/site-packages/pythran/pythonic/core.hpp:41:
    In file included from ../../../../../../../private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-build-env-yj77gq5q/overlay/lib/python3.10/site-packages/pythran/pythonic/types/assignable.hpp:4:
    ../../../../../../../private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-build-env-yj77gq5q/overlay/lib/python3.10/site-packages/pythran/pythonic/include/types/assignable.hpp:15:12: warning: local variable 't' will be copied despite being returned by name [-Wreturn-std-move]
        return t;
               ^
    scipy/stats/_stats_pythran.cpp:640:308: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::types::as_const<(anonymous namespace)::pythonic::types::array_base<(anonymous namespace)::pythonic::types::ndarray<long, (anonymous namespace)::pythonic::types::array_base<long, 1, (anonymous namespace)::pythonic::types::tuple_version>>, 1, (anonymous namespace)::pythonic::types::tuple_version>>' requested here
              typename pythonic::assignable_noescape<decltype(std::get<0>(pythonic::types::as_const(pythonic::numpy::functor::nonzero{}(pythonic::types::as_const(deltax)(j,pythonic::types::contiguous_slice(pythonic::builtins::None,pythonic::builtins::None))))))>::type id_nonzero = std::get<0>(pythonic::types::as_const(pythonic::numpy::functor::nonzero{}(pythonic::types::as_const(deltax)(j,pythonic::types::contiguous_slice(pythonic::builtins::None,pythonic::builtins::None)))));
                                                                                                                                                                                                                                                                                                                       ^
    scipy/stats/_stats_pythran.cpp:1056:84: note: in instantiation of function template specialization '(anonymous namespace)::__pythran__stats_pythran::siegelslopes::operator()<(anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long>>, (anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long>>, (anonymous namespace)::pythonic::types::str>' requested here
                                    auto res = __pythran__stats_pythran::siegelslopes()(y, x, method);
                                                                                       ^
    ../../../../../../../private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-build-env-yj77gq5q/overlay/lib/python3.10/site-packages/pythran/pythonic/include/types/assignable.hpp:15:12: note: call 'std::move' explicitly to avoid copying
        return t;
               ^
               std::move(t)
    ../../../../../../../private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-build-env-yj77gq5q/overlay/lib/python3.10/site-packages/pythran/pythonic/include/types/assignable.hpp:15:12: warning: local variable 't' will be copied despite being returned by name [-Wreturn-std-move]
        return t;
               ^
    scipy/stats/_stats_pythran.cpp:644:146: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::types::as_const<(anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::sub, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::mul, (anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long>> &, (anonymous namespace)::pythonic::types::broadcast<double, double>>, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::mul, (anonymous namespace)::pythonic::types::broadcast<double, double>, (anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long>> &>>>' requested here
                pythonic::builtins::list::functor::append{}(intercepts, pythonic::numpy::functor::median{}(pythonic::operator_::div(pythonic::types::as_const(pythonic::operator_::sub(pythonic::operator_::mul(y, pythonic::types::as_const(x)[j]), pythonic::operator_::mul(pythonic::types::as_const(y)[j], x)))[id_nonzero], pythonic::types::as_const(deltax)[pythonic::types::make_tuple(j, id_nonzero)])));
                                                                                                                                                     ^
    scipy/stats/_stats_pythran.cpp:1056:84: note: in instantiation of function template specialization '(anonymous namespace)::__pythran__stats_pythran::siegelslopes::operator()<(anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long>>, (anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long>>, (anonymous namespace)::pythonic::types::str>' requested here
                                    auto res = __pythran__stats_pythran::siegelslopes()(y, x, method);
                                                                                       ^
    ../../../../../../../private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-build-env-yj77gq5q/overlay/lib/python3.10/site-packages/pythran/pythonic/include/types/assignable.hpp:15:12: note: call 'std::move' explicitly to avoid copying
        return t;
               ^
               std::move(t)
    ../../../../../../../private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-build-env-yj77gq5q/overlay/lib/python3.10/site-packages/pythran/pythonic/include/types/assignable.hpp:15:12: warning: local variable 't' will be copied despite being returned by name [-Wreturn-std-move]
        return t;
               ^
    scipy/stats/_stats_pythran.cpp:644:146: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::types::as_const<(anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::sub, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::mul, (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long>> &, (anonymous namespace)::pythonic::types::broadcast<float, float>>, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::mul, (anonymous namespace)::pythonic::types::broadcast<float, float>, (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long>> &>>>' requested here
                pythonic::builtins::list::functor::append{}(intercepts, pythonic::numpy::functor::median{}(pythonic::operator_::div(pythonic::types::as_const(pythonic::operator_::sub(pythonic::operator_::mul(y, pythonic::types::as_const(x)[j]), pythonic::operator_::mul(pythonic::types::as_const(y)[j], x)))[id_nonzero], pythonic::types::as_const(deltax)[pythonic::types::make_tuple(j, id_nonzero)])));
                                                                                                                                                     ^
    scipy/stats/_stats_pythran.cpp:1072:84: note: in instantiation of function template specialization '(anonymous namespace)::__pythran__stats_pythran::siegelslopes::operator()<(anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long>>, (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long>>, (anonymous namespace)::pythonic::types::str>' requested here
                                    auto res = __pythran__stats_pythran::siegelslopes()(y, x, method);
                                                                                       ^
    ../../../../../../../private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-build-env-yj77gq5q/overlay/lib/python3.10/site-packages/pythran/pythonic/include/types/assignable.hpp:15:12: note: call 'std::move' explicitly to avoid copying
        return t;
               ^
               std::move(t)
    3 warnings generated.
```

So it's three times the same warning from a single place in `pythran/pythonic/include/types/assignable.hpp`. That would be nice to resolve.

--- a/pythran/pythonic/include/types/assignable.hpp
+++ b/pythran/pythonic/include/types/assignable.hpp
@@ -12,7 +12,7 @@ namespace types
   template <class T>
   constexpr T as_const(T &&t) noexcept
   {
-    return t;
+    return std::forward<T>(t);
   }
 
   // Pass all scalars by value when called through pythonic::types::call

